peekingduck/training/src/callbacks/base.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2023 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+from typing import (
+    List,
+    Any,
+    TYPE_CHECKING,
+    Union,
+)  # This solves circular import type hinting.
+from omegaconf import DictConfig
+import src.callbacks as Callbacks
+
+if TYPE_CHECKING:
+    from src.trainer import Trainer
+
+
+def init_callbacks(callbacks: List[Union[DictConfig, str]]) -> List:
+    """Method to initialise the callbacks.
+    Args:
+        callbacks: The list of callbacks from configs->callbacks->classification/detection/segmentation.
+    """
+    cb_list = []
+    for cb in callbacks:
+        if type(cb) is DictConfig:
```

Comment:
Consider using `isinstance` for type checking.

The reason is subclass can be also checked, so it is a stronger condition than using `type` equality check.

See [here](https://stackoverflow.com/questions/30049410/why-use-isinstance-instead-of-type).